code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -37,26 +37,98 @@
 	icon_resting = "cat_rest"
 	gender = FEMALE
 	gold_core_spawnable = CHEM_MOB_SPAWN_INVALID
+	var/list/family = list()
+	var/memory_saved = 0
+	var/list/children = list() //Actual mob instances of children
+	var/cats_deployed = 0
+
+/mob/living/simple_animal/pet/cat/Runtime/New()
+	Read_Memory()
+	..()
+
+/mob/living/simple_animal/pet/cat/Runtime/Life()
+	if(!cats_deployed && ticker.current_state >= GAME_STATE_SETTING_UP)
+		Deploy_The_Cats()
+	if(!stat && ticker.current_state == GAME_STATE_FINISHED && !memory_saved)
+		Write_Memory()
+	..()
+
+/mob/living/simple_animal/pet/cat/Runtime/make_babies()
+	var/mob/baby = ..()
+	if(baby)
+		children += baby
+		return baby
+
+/mob/living/simple_animal/pet/cat/Runtime/death()
+	if(!memory_saved)
+		Write_Memory(1)
+	..()
+
+/mob/living/simple_animal/pet/cat/Runtime/proc/Read_Memory()
+	var/savefile/S = new /savefile("data/npc_saves/Runtime.sav")
+	S["family"] 			>> family
+
+	if(isnull(family))
+		family = list()
+
+/mob/living/simple_animal/pet/cat/Runtime/proc/Write_Memory(dead)
+	var/savefile/S = new /savefile("data/npc_saves/Runtime.sav")
+	family = list()
+	if(!dead)
+		for(var/mob/living/simple_animal/pet/cat/kitten/C in children)
+			if(istype(C,type) || C.stat || !C.z || !C.butcher_results)
+				continue
+			if(C.type in family)
+				family[C.type] += 1
+			else
+				family[C.type] = 1
+	S["family"]				<< family
+	memory_saved = 1
+
+/mob/living/simple_animal/pet/cat/Runtime/proc/Deploy_The_Cats()
+	cats_deployed = 1
+	for(var/cat_type in family)
+		if(family[cat_type] > 0)
+			for(var/i in 1 to min(family[cat_type],100)) //Limits to about 500 cats, you wouldn't think this would be needed (BUT IT IS)
+				new cat_type(loc)
+
 
 /mob/living/simple_animal/pet/cat/handle_automated_action()
 	..()
+	if(prob(1))
+		custom_emote(1, pick("stretches out for a belly rub.", "wags its tail.", "lies down."))
+		icon_state = "[icon_living]_rest"
+		resting = 1
+		update_canmove()
+	else if (prob(1))
+		custom_emote(1, pick("sits down.", "crouches on its hind legs.", "looks alert."))
+		icon_state = "[icon_living]_sit"
+		resting = 1
+		update_canmove()
+	else if (prob(1))
+		if (resting)
+			custom_emote(1, pick("gets up and meows.", "walks around.", "stops resting."))
+			icon_state = "[icon_living]"
+			resting = 0
+			update_canmove()
+		else
+			custom_emote(1, pick("grooms its fur.", "twitches its whiskers.", "shakes out its coat."))
 
 	//MICE!
 	if(eats_mice && loc && isturf(loc) && !incapacitated())

```

Comment:
Actually, `loc && isturf(loc)` is redundant. You can change this to `if(eats_mice && isturf(loc) && !incapacitated())`.